operators/endpointmetrics/controllers/observabilityendpoint/ocp_monitoring_config.go

```diff
@@ -494,10 +494,10 @@ func unset(ctx context.Context, client client.Client, ns string) error {
 	return err
 }
 
-// revertClusterMonitoringConfig reverts the configmap cluster-monitoring-config and relevant resources
+// RevertClusterMonitoringConfig reverts the configmap cluster-monitoring-config and relevant resources
 // (observability-alertmanager-accessor and hub-alertmanager-router-ca) for the openshift cluster monitoring stack.
-func revertClusterMonitoringConfig(ctx context.Context, client client.Client) error {
-	log.Info("revertClusterMonitoringConfig called")
+func RevertClusterMonitoringConfig(ctx context.Context, client client.Client) error {
+	log.Info("RevertClusterMonitoringConfig called")
 
 	// try to retrieve the current configmap in the cluster
 	found := &corev1.ConfigMap{}
```

Comment:
This is perhaps a more involved change, but instead of relying on a backup of the entire object, making a change and reverting to that - we only ever add `additionalAlertmanagerConfig`, should we change the revert logic to find the index in the config slice and just delete it instead? This means if the backup doesn't exist, but somehow endpointoperator has made the change it will always remove it if it finds it? 

We have the hub alertmanager svc hardcoded, so we could always use that to find which entry this endpoint operator is responsible for? 

As an extension, to avoid double alerting, we could also error if that list is >0 (user editing of this config map is discouraged by CMO, so we can be certain that array should only be one item long).

Reply:
Yes, I think the function itself needs some work and improvement but I think it would be extending the scope of this jira/problem to do so here.

I have started a basic refactor in https://github.com/stolostron/multicluster-observability-operator/pull/1575 which reduces complexity in the first place and should essentially be a no-op right now for behaviour. I would lean towards us merging both of these PRs first before making any radical changes to how we handle the config.